src/curl_jolokia.c

```diff
@@ -0,0 +1,1100 @@
+    return -1;
+  return strncmp(pbeancfg->mbean_match, db->json_key, db->json_key_len);
+}
+
+/*
+ * Aligning data with Configs & Sending it
+ */
+
+static int cjo_get_type(cjo_attribute_t *attribute) {
+  const data_set_t *ds;
+
+  assert(attribute->type != NULL);
+  ds = plugin_get_ds(attribute->type);
+  if (ds == NULL) {
+    static char type[DATA_MAX_NAME_LEN] = "!!!invalid!!!";
+
+    if (strcmp(type, attribute->type) != 0) {
+      ERROR("curl_jolokia plugin: Unable to look up DS type \"%s\".",
+            attribute->type);
+      sstrncpy(type, attribute->type, sizeof(type));
+    }
+
+    return -1;
+  } else if (ds->ds_num > 1) {
+    static c_complain_t complaint = C_COMPLAIN_INIT_STATIC;
+
+    c_complain_once(
+        LOG_WARNING, &complaint,
+        "curl_jolokia plugin: The type \"%s\" has more than one data source. "
+        "This is currently not supported. I will return the type of the "
+        "first data source, but this will likely lead to problems later on.",
+        attribute->type);
+  }
+
+  return ds->ds[0].type;
+}
+
+static void cjo_submit(cjo_t *db) /* {{{ */
+{
+  llentry_t *cfgptr;
+  char buffer[db->max_value_len + 1];
+  value_t ret_value;
+  char *host;
+  int status;
+  int i;
+  int ds_type;
+  cjo_attribute_t *curr_attribute;
+
+  if ((db->match_this_bean == NULL) || (db->attribute_pool_used == 0))
+    return; /* nothing to do yet. */
+
+  for (i = 0; i < db->attribute_pool_used; i++) {
+    value_list_t vl = VALUE_LIST_INIT;
+
+    db->curr_attribute = &db->attributepool[i];
+    cfgptr = llist_search_custom(db->match_this_bean->attributes,
+                                 cjo_cfg_compare_attribute, db);
+    if (cfgptr == NULL) {
+      char attribute[db->curr_attribute->json_name_len + 1];
+
+      memcpy(attribute, db->curr_attribute->json_name,
+             db->curr_attribute->json_name_len);
+      attribute[db->curr_attribute->json_name_len] = '\0';
+      ERROR("curl_jolokia plugin: failed to locate attribute [%s:\"%s\"]",
+            db->match_this_bean->bean_name, attribute);
+
+      continue; /* we don't know this! */
+    }
+    curr_attribute = cfgptr->value;
+
+    /* Create a null-terminated version of the string. */
+    ds_type = cjo_get_type(curr_attribute);
+    if (ds_type == -1) {
+      char attribute[db->curr_attribute->json_name_len + 1];
+
+      memcpy(attribute, db->curr_attribute->json_name,
+             db->curr_attribute->json_name_len);
+      attribute[db->curr_attribute->json_name_len] = '\0';
+
+      ERROR("curl_jolokia plugin: failed to map type for [%s:%s:%s]",
+            db->match_this_bean->bean_name, attribute, curr_attribute->type);
+      continue;
+    }
+
+    memset(&ret_value, 0, sizeof(ret_value));
+
+    memcpy(buffer, db->curr_attribute->json_value,
+           db->curr_attribute->json_value_len);
+
+    buffer[db->curr_attribute->json_value_len] = '\0';
+
+    status = parse_value(buffer, &ret_value, ds_type);
```

Comment:
Only assign the return value to a variable if you need it later, e.g. to log it. Otherwise, do:

```c
if (parse_value(/* … */) != 0 {
  /* … */
}
```